src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -187,6 +201,163 @@ class UserService(protected val userInfo: UserInfo, dataSource: DataSource, prot
     }
   }
 
+  def listGroupMembers(groupName: String) = {
+    asAdmin {
+      dataSource.inFutureTransaction() { txn =>
+        Future {
+          containerDAO.authDAO.loadGroup(RawlsGroupRef(RawlsGroupName(groupName)), txn) match {
+            case None => RequestComplete(ErrorReport(StatusCodes.NotFound, s"Group ${groupName} does not exist"))
+            case Some(group) =>
+              val memberUsers = group.users.map(u => containerDAO.authDAO.loadUser(u, txn).get.userEmail.value)
+              val memberGroups = group.subGroups.map(g => containerDAO.authDAO.loadGroup(g, txn).get.groupEmail.value)
+              RequestComplete(StatusCodes.OK, UserList((memberUsers ++ memberGroups).toSeq))
+          }
+        }
+      }
+    }
+  }
+
+  def createGroup(groupRef: RawlsGroupRef) = {
+    asAdmin {
+      dataSource.inTransaction() { txn =>
+        containerDAO.authDAO.loadGroup(groupRef, txn) match {
+          case Some(_) => Future.successful(RequestComplete(ErrorReport(StatusCodes.Conflict, s"Group ${groupRef.groupName} already exists")))
+          case None =>
+            containerDAO.authDAO.createGroup(RawlsGroup(groupRef.groupName, RawlsGroupEmail(gcsDAO.toGoogleGroupName(groupRef.groupName)), Set.empty[RawlsUserRef], Set.empty[RawlsGroupRef]), txn)
+            gcsDAO.createGoogleGroup(groupRef) map { _ => RequestComplete(StatusCodes.Created) }
+        }
+      }
+    }
+  }
+
+  def deleteGroup(groupRef: RawlsGroupRef) = {
+    asAdmin {
+      dataSource.inTransaction() { txn =>
+        withGroup(groupRef) { group =>
+          containerDAO.authDAO.deleteGroup(groupRef, txn)
+          gcsDAO.deleteGoogleGroup(groupRef) map { _ => RequestComplete(StatusCodes.OK) }
+        }
+      }
+    }
+  }
+
+  //ideally this would probably just return the already loaded users to avoid loading twice
+  def allMembersExist(memberList: RawlsGroupMemberList): Boolean = {
+    dataSource.inTransaction() { txn =>
+      memberList.userEmails.foreach { user =>
+        containerDAO.authDAO.loadUserByEmail(user, txn).getOrElse(return false)
+      }
+      memberList.subGroupEmails.foreach { subGroup =>
+        containerDAO.authDAO.loadGroupByEmail(subGroup, txn).getOrElse(return false)
+      }
+      true
```

Comment:
:-1: return statements, this should map each list of emails to load the object, concatenate those 2 lists then return  !list.exists(None)